### PR TITLE
Add hdmf-zarr recipe

### PR DESCRIPTION
**Pull request by mavaylon1**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by synapticarbors** [comment]
I've added an upstream PR to build the needed pynwb 2.0.1: https://github.com/conda-forge/pynwb-feedstock/pull/42

**Comment by mavaylon1** [comment]
@conda-forge/help-python ready for review

File: recipes/hdmf_zarr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: 281cfcd1581ed8e101953015179aedc01e55740980540f7b0e43f67110fa44f5
+  
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  noarch: python
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (dopplershift) [inline recipes/hdmf_zarr/meta.yaml:19]: This is pure Python and doesn't seem to need a compiler:
```suggestion
```

Comment (synapticarbors) [inline recipes/hdmf_zarr/meta.yaml:19]: It does not look like a compiler is needed to build this package, so you can remove this and the `build` section.

File: recipes/hdmf_zarr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  host:
+    - python >=3.7
+    - pip
+  run:
```
Comment (synapticarbors) [inline recipes/hdmf_zarr/meta.yaml:23]: You should follow the version pinning in the `setup.py`, which isn't specifying exact versions.

File: recipes/hdmf_zarr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - python >=3.7
+    - hdmf ==3.4.0
+    - zarr ==2.11.0
+    - numcodecs ==0.9.1
+    - pynwb ==2.0.0
+    - setuptools
+
+test:
```
Comment (synapticarbors) [inline recipes/hdmf_zarr/meta.yaml:31]: I would recommend adding the `pip check` test back in.

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a few minor tweaks and this is good to go.

File: recipes/hdmf_zarr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - {{ compiler('c') }}
+  host:
+    - python >=3.7
+    - pip
+  run:
+    - python >=3.7
+    - hdmf ==3.4.0
+    - zarr ==2.11.0
+    - numcodecs ==0.9.1
+    - pynwb ==2.0.0
```
[the remark this answers is not in the document]
Comment (mavaylon1) [reply recipes/hdmf_zarr/meta.yaml:26]: These just match the reqs for what is currently supported. HDMF has regular updates that may or may not clash with the other packages. I chose the exact versions and prefer to update as we confirm what works in development.

[the remark this answers is not in the document]
Comment (dopplershift) [reply recipes/hdmf_zarr/meta.yaml]: Then why does upstream list it with `>=`?

[the remark this answers is not in the document]
Comment (dopplershift) [reply recipes/hdmf_zarr/meta.yaml]: Talking about here: https://github.com/hdmf-dev/hdmf-zarr/blob/48ab39f6cd8950a572dbff378ed104caf5a60019/setup.py#L19-L24

I was just proposing making the PyPI and conda-forge packages to have identical metadata.

[the remark this answers is not in the document]
Comment (mavaylon1) [reply recipes/hdmf_zarr/meta.yaml]: What is upstream?

[the remark this answers is not in the document]
Comment (mavaylon1) [reply recipes/hdmf_zarr/meta.yaml]: Ahhh I see

File: recipes/hdmf_zarr/meta.yaml (hunk 3 above)
Comment (dopplershift) [reply recipes/hdmf_zarr/meta.yaml:29]: Can you add as was requested here?
```suggestion
test:
  commands:
    - pip check
```

**Comment by mavaylon1** [comment]
@conda-forge-bot ready for review

**Review by mavaylon1 — COMMENTED** [review]
Done

**Review by dopplershift — APPROVED** [review]

**Pull request merged**